### PR TITLE
PWN-3724 - Fix SOL price on Derivable Accounts

### PR DESCRIPTION
**Pull request by kuznetsov**
### Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3724

### Description of Work

Since token rates are requested on `HomeFragment` we don't have it on `DerivableAccountsFragment`. I requested `SOL` price only on `DerivableAccountsFragment` to calculate total USD balances.

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt [hunk 1]
```diff
@@ -36,6 +36,13 @@ class TokenPricesCoinGeckoRepository(
         loadPrices(tokenSymbols, targetCurrency)
     }
 
+    override suspend fun getTokenPriceBySymbol(
+        tokenSymbol: TokenSymbol,
+        targetCurrency: String
+    ): TokenPrice {
+        return loadPrices(listOf(tokenSymbol), targetCurrency).first()
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt:43]: Are we able to load a price for a single token?

Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt:43]: Actually there are methods for single prices, I just used the implemented ones

Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt]: Actually, you're right. If we're sending request for one token price, why not requesting all and cache them before home

Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt]: But if we want to load and cache all user tokens prices on DerivableAccount we should make 2 bigger requests and make caching job, screen could be overloaded

File: app/src/main/java/org/p2p/wallet/restore/interactor/SecretKeyInteractor.kt [hunk 2]
```diff
@@ -56,13 +60,18 @@ class SecretKeyInteractor(
             return@withContext result
         }
 
-    private fun mapDerivableAccounts(
+    private suspend fun mapDerivableAccounts(
         accounts: List<Account>,
         balances: List<Pair<String, BigInteger>>,
         path: DerivationPath
     ) = accounts.mapNotNull { account ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/restore/interactor/SecretKeyInteractor.kt:67]:
```suggestion
    ) : List< DerivableAccount> = accounts.mapNotNull { account ->
```

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**